app.py

```diff
@@ -92,15 +99,17 @@ def loginNLU():
     try:
         serviceURL = request.form['host']
         apiKey = request.form['password']
-        
+
         # Create the authenticator.
         authenticator = IAMAuthenticator(apiKey)
         natural_language_understanding = NaturalLanguageUnderstandingV1(
             version='2021-03-25',
             authenticator=authenticator)
-            
-        natural_language_understanding.set_service_url(serviceURL)
 
+        if serviceURL.find("https://") == -1:
```

Comment:
@VidyasagarMSC 

I would make sure to trim serviceURL (`.strip()`) and then use `. startswith()` to check if the `.lower()` version starts with `http://` or `https://` before adding `https://`

string to test with:
`acme.com`
`HTTPS://acme.com`
`HTTP://acme.com`
`FTP://acme.com`
`http://acme.com`
`https://acme.com`